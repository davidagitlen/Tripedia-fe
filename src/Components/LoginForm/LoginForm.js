import React, { useState } from "react";
import "./LoginForm.scss";
import { NavLink } from "react-router-dom";
import banner from "../../images/banner.jpg";

const LoginForm = () => {
  const [loginState, handleForm] = useState({
    email: "",
    password: ""
  });
  const { email, password } = loginState;
  const isEnabled = email && password;

  const handleChange = e => {
    handleForm({ ...loginState, [e.target.name]: e.target.value });
  };

  return (
    <main>
      <img
        className="banner"
        src={banner}
        alt="Tripedia for all your travel planning needs"
      />
      <form className="login_form">
        <h2>Login Form</h2>
        <input
          className="email_input"
          type="email"
          name="email"
          placeholder="email"
          value={email}
          onChange={handleChange}
        ></input>
        <input
          type="password"
          name="password"
          className="password_input"
          placeholder="password"
          value={password}
          onChange={handleChange}
        ></input>
        <NavLink to="/" className="button">
          <button disabled={!isEnabled} className="login_button" type="submit">
            Login
          </button>
        </NavLink>
        <NavLink to="/create_account">
          <p className="create_account_link">Create Account</p>
        </NavLink>
      </form>
    </main>
  );
};

export default LoginForm;
