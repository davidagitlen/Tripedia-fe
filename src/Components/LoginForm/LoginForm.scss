.login-form__container {
  padding-left: 125px;
  padding-right: 125px;
  padding-top: 125px;
  background-color: rgba(32, 113, 120, .3);
  height: 100vh;

  .banner {
    width: 100%;
    height: 350px;
    border-radius: 40px;
  }

  .login__form {
    height: 250px;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
    padding-left: 150px;
    padding-right: 150px;
    margin-top: 15px;

    a {
      width: 100%;
      display: flex;
      justify-content: center;
      text-decoration: none;
    }
  }

  input {
    width: 100%;
    height: 40px;
    border-radius: 40px;
    border: 2px solid grey;
    text-align: center;
  }

  .button__disabled {
    width: 50%;
    border-radius: 40px;
    height: 40px;
    background-color: rgba(128, 128, 128, .3)
  }

  .button__enabled {
    width: 50%;
    border-radius: 40px;
    height: 40px;
    background-color: #FF9766;
    color: #207178;
    font-weight: bold;
    font-size: 20px;
    border: 2px solid #207178;
  }
}

@media screen and (min-width: 300px) {
  .login-form__container {
    padding-left: 50px;
    padding-right: 50px;
    padding-top: 40px;

    .login__form {
      padding-left: 50px;
      padding-right: 50px;
    }

    .banner {
      height: 200px;
    }
  }
}

@media screen and (min-width: 800px) {
  .login-form__container {
    padding-left: 100px;
    padding-right: 100px;
    padding-top: 100px;

    .login__form {
      padding-left: 200px;
      padding-right: 200px;
    }

    .banner {
      height: 240px;
    }
  }
}

@media screen and (min-width: 1200px) {
  .login-form__container {
    padding-left: 300px;
    padding-right: 300px;

    .banner {
      height: 350px;
    }
  }
}

@media screen and (min-width: 1800px) {
  .login-form__container {
    padding-left: 400px;
    padding-right: 400px;
  }
}