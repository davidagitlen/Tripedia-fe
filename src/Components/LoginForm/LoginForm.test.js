import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './LoginForm';

describe('LoginForm', () => {

  it('should pass a dummy test', () => {
    expect(true).toEqual(true);
  });

});