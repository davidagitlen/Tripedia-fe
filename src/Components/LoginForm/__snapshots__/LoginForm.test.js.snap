// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match the wrapper with data passed in 1`] = `
<main
  className="login-form__container"
>
  <img
    alt="Tripedia for all your travel planning needs"
    className="banner"
    src="banner.png"
  />
  <form
    className="login__form"
  >
    <input
      name="email"
      onChange={[Function]}
      placeholder="example@email.com"
      type="email"
      value="g@g.com"
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="enter password"
      type="password"
      value="pass"
    />
    <NavLink
      to="/map"
    >
      <button
        className="button__enabled"
        onClick={[Function]}
        type="submit"
      >
        Login
      </button>
    </NavLink>
    <NavLink
      to="/create_account"
    >
      <p>
        Create Account
      </p>
    </NavLink>
  </form>
</main>
`;
